frontend/src/App.tsx

```diff
@@ -3,19 +3,106 @@ import {BrowserRouter as Router, Routes, Route} from "react-router-dom";
 import Footer from "./components/Footer/Footer.tsx";
 import Header from "./components/Header/Header.tsx";
 import Applications from "./components/Applications/Applications.tsx";
+import ApplicationForm from "./components/CreateForm/ApplicationForm.tsx";
+import { useState } from "react";
+import {Application} from "./components/Applications/Application.ts";
 
 export default function App() {
+    const [reloadRotate, setReloadRotate] = useState<boolean>(false);
+    const [addRotate, setAddRotate] = useState<boolean>(false);
+    const [reloadKey, setReloadKey] = useState<number>(0);
+    const [formData, setFormData] = useState<{
+        applicationId?: string;
+        initialData: { companyName: string; status: string };
+    } | null>(null);
+    const [showForm, setShowForm] = useState<boolean>(false);
+    const [loading, setLoading] = useState<boolean>(false);
+    const [selectedApplication, setSelectedApplication] = useState<Application | null>(null);
 
-  return (
+    const disableButtons = loading || showForm || selectedApplication !== null;
+
+    function handleReload() {
+        setReloadRotate(true);
+        setReloadKey(prevKey => prevKey + 1);
+        setTimeout(() => {
+            setReloadRotate(false);
+        }, 250);
+    }
+
+    function handleAdd() {
+        setAddRotate(true);
+
+        setTimeout(() => {
+            setAddRotate(false);
+        }, 250);
+        setFormData({
+            applicationId: undefined, // Keine ID, da es sich um eine neue Bewerbung handelt
+            initialData: {
+                companyName: '', // Leeres Feld für Firmenname
+                status: '', // Leeres Feld für Status
+            },
+        });
+        setShowForm(true); // Formular anzeigen
+    }
+
+    return (
 
         <Router>
             <Header/>
 
             <Routes>
-                <Route path="/" element={<Applications />} />
+                <Route path="/" element={
+                    <Applications
+                        reloadKey={reloadKey}
+                        setFormData={setFormData} // Zum Bearbeiten von Bewerbungen
+                        setShowForm={setShowForm}
+                        setLoading={setLoading}
+                        loading={loading}
+                        setSelectedApplication={setSelectedApplication}
+                        selectedApplication={selectedApplication}
+                    />
+                } />
             </Routes>
-            <Footer/>
-
+            <Footer
```

Comment:
so much logik for the footer? maby rename component

Reply:
It is because of a big trouble with the visibilty of the buttons when the form-overlay is switched on....